File: Week 2/QuangKhaik14PM01-cs50-problems-2023-x-readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

// Nguyên mẫu các hàm
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Nhận văn bản từ người dùng
    string text = get_string("Text: ");

    // Đếm số lượng chữ cái, từ và câu
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Tính chỉ số Coleman-Liau
    float L = (float)letters / words * 100;
    float S = (float)sentences / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // In kết quả
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", round(index));
    }

    return 0;
}

// Hàm đếm số lượng chữ cái trong văn bản
int count_letters(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

// Hàm đếm số lượng từ trong văn bản
int count_words(string text)
{
    int count = 1; // Khởi tạo số từ là 1 vì có ít nhất một từ
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            count++;
        }
    }
    return count;
}

// Hàm đếm số lượng câu trong văn bản
int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}

File: week6_Python/problem-set/sentimental-readability/readability.py
# TODO

def main():

    sentence = input("Text: ")

    # Number of letters, words and sentences in the string.
    letters = count_letters(sentence)
    words = count_words(sentence)
    sentences = count_sentences(sentence)

    # Average number of letters and stentences per 100 words in text.
    avg_letters = (100 / float(words)) * float(letters)
    avg_sentences = (100 / float(words)) * float(sentences)

    # Coleman-Liau index.
    index = calculate_index(avg_letters, avg_sentences)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


# Count the number of letters
def count_letters(sentence):


    letters = 0

    for character in sentence:
        if character.isalpha():
            letters += 1

    return letters

# Count the number of words
def count_words(sentence):

    return len(sentence.split())


# Count the number of sentences
def count_sentences(sentence):

    sentences = 0

    for character in sentence:
        if character in [".", "!", "?"]:
            sentences += 1

    return sentences


def calculate_index(avg_letters, avg_sentences):
    return round(0.0588 * avg_letters - 0.296 * avg_sentences - 15.8)


if __name__ == "__main__":
    main()
